tests/conftest.py

```diff
@@ -44,9 +46,93 @@ def redis(redis_adapter):
 
 @pytest.fixture(scope='function')
 def api():
-    return Api(api_url='https://api.localhost.ai', data_api_url='https://data.localhost.ai')
+    return Api(
+        api_url='https://api.localhost.ai',
+        data_api_url='https://data.localhost.ai',
+        api_key='',
+        app_name=''
+    )
+
+
+@pytest.fixture(scope='function')
+def settings():
+    return Settings(
+        APP_KEY='provider.app-name',
+        CACHE_URL='redis://localhost:6379'
+    )
+
+
+@pytest.fixture(scope='function', autouse=True)
+def patch_settings(settings, mocker):
+    settings_path = 'corva.settings.SETTINGS'
+
+    mocker.patch.multiple(
+        settings_path,
+        APP_KEY=settings.APP_KEY,
+        CACHE_URL=settings.CACHE_URL
+    )
+    yield
+
+
+@pytest.fixture(scope='session')
+def raw_stream_event() -> str:
+    with open(DATA_PATH / 'stream_event.json') as stream_event:
+        return stream_event.read()
 
 
 class ComparableException(Exception):
     def __eq__(self, other):
         return type(self) is type(other) and self.args == other.args
+
+
+class StreamDataMixer:
```

Comment:
can we simplify the test's logic and not create classes with a specific logic of how tests should work?
It would be much simpler to understand what is happening in the test inside the scope of itself, rather than figure out the logic defined elsewhere

Reply:
Stream events have many different fields. Without StreamDataMixer it'd be hard to create custom StreamEvent object in every test

Reply:
Maybe it would be harder to create, but it would be easier to read and change. Engineers read code more than write. We need to worry about us reading this code, not how hard it would be to write it

Reply:
Q What current mixer logic does?
A It constructs and returns a pydantic model, providing required parameters to constuctor (without those params model raises an exception). We can override parameters as needed, otherwise just call the function without arguments to get the model.

Q Why do we need to generate models?
A We need to generate models (such as events) and cast them to json, because this is what will be received in real-world from kafka

Cool scenario would be the support of pydantic models in `mixer` [library](https://pypi.org/project/mixer/), so there is no need to write classes like this. But there is no support.

We are left with a few options then:

1. Delete the mixer from repository. Insert raw events in every test, modifying some fields as needed. Raw events look like this: `'[{"app_key": null, "records": [{"timestamp": null, "asset_id": 0, "company_id": 0, "version": 0, "measured_depth": null, "collection": "wits.completed", "data": {"hole_depth": null, "weight_on_bit": null, "state": null}}], "metadata": {"app_stream_id": 0, "source_type": null, "apps": {"provider.app-name": {"app_connection_id": 0}}}, "asset_id": 0}]'`.
Concern - it may be hard to maintain such raw events (in case model changes).

2. Use some library like [factory_boy](https://factoryboy.readthedocs.io/en/stable/), which would require creating factory_boy models (replicating pydantic models).

3. Current mixer class.

Reply:
The example you provided is not big

Reply:
what example?

if I understand correctly, you say, that option 1 would be the best?

Reply:
will rewrite the tests as in option #1